Update target framework and package versions and add docker support

meysam opened the pull request:
The target framework in the `Authlete` and `Authlete.Tests` projects has been updated from `netstandard1.4` and `netcoreapp2.0` respectively to `net8.0`. 
Additionally, the versions of several packages have been upgraded for improved compatibility and latest features. 
A `global.json` file has also been added to specify `sdk version` and `options`.

meysam committed:
Update target framework and package versions

The target framework in the Authlete and Authlete.Tests projects has been updated from netstandard1.4 and netcoreapp2.0 respectively to net8.0. Additionally, the versions of several packages have been upgraded for improved compatibility and latest features. A global.json file has also been added to specify sdk version and options.

meysam committed:
Add Docker support for building and testing

The project now supports building and running inside a Docker container. This includes details on how to build the Docker image with verbose build output, run the container, and retrieve test results from the Docker container. This change also adds a Dockerfile and .dockerignore file to the project. Updates were also made to the Authlete.Tests project file.

meysam committed:
Remove .NET publishing steps from Dockerfile

The Dockerfile has been simplified by removing the steps related to .NET application publishing and execution. The .NET publishing, final image stage, and command lines have been deleted, leaving only the commands for copying the test results.

meysam committed:
Update CHANGES.md with 1.6.0 version details.

meysam committed:
Update Authlete package version to 1.6.0.

hefa commented on README.md:
```diff
@@ -344,6 +344,54 @@ How To Test
+- `-t authlete-test` tags the built Docker image as `authlete-test`.
+- `--progress=plain` displays the build output in plain text, providing detailed information during the build process.
+- `--no-cache` ensures that Docker does not use any cached layers from previous builds, forcing all steps to be re-executed.
+- `--target build` specifies that the Docker build process should stop after completing the `build` stage. This is particularly useful if you want to run tests without deploying the final application as part of the build process.
```
There is only one stage in the Dockerfile, so this is not necessary

hefa commented on Dockerfile:
```diff
@@ -0,0 +1,27 @@
+RUN dotnet build "Authlete/Authlete.csproj" -c Release -o /app/build
+
+# Run tests and generate TRX test results
+RUN dotnet test "Authlete.Tests/Authlete.Tests.csproj" -c Release --no-restore --logger "trx;LogFileName=test_results.trx"
```
Can the `dotnet test` command output the results to the console? Then there's no need to write to files and retrieve them from inside the container.

meysam committed:
Simplify Docker build process and update documentation

Removed the 'target build' stage from the Docker file and related command line argument in README.md. The test results are now displayed in console with detailed verbosity, eliminating the previous step of copying test results to a known location.
